src/pages/library/common.js

```diff
@@ -86,8 +99,132 @@ const FilterSection = ({ onTagFilter, labels, selectedTags, labelRenderer, listD
     }, labelsToDisplay),
     _.size(labels) > numLabelsToRender && h(Link, {
       style: { display: 'block', textAlign: 'center' },
-      onClick: () => setShowAll(!showAll)
-    }, [`See ${showAll ? 'less' : 'more'}`])
+      onClick: () => {
+        setShowAll(!showAll)
+        setFilterChanges({})
+
+        const _tagFilters = {}
+        _.forEach(label => {
+          const lowerTag = _.toLower(label)
+          _tagFilters[lowerTag] = listDataByTag[lowerTag]
+        }, labels)
+        setFilteredTags(_tagFilters)
+      }
+    }, [`See more`]),
+    showAll && h(Modal, {
+      title: ``,
+      width: '100%',
+      showButtons: true,
+      okButton: 'Apply Filters',
+      styles: {
+        modal: { maxWidth: 900, padding: 30 },
+        buttonRow: { width: '100%', borderTop: `6px solid ${colors.dark(0.1)}`, paddingTop: 20 }
+      },
+      onDismiss: () => {
+        setShowAll(false)
+        _.forEach(onTagFilter, filterChanges)
+        setFilterChanges({})
+        setFilterSearchText('')
+        setFilteredLabels(labels)
+        setFilterType(filterOptions.alpha)
+      }
+    }, [
```

Comment:
How would you feel about factoring out the filter index bar into its own component (e.g. PartitionBar)?
It could support events like:

1. onPartition
2. onTopPartition (or onSpecial)
3. onSearch
4. onClear

I think this would help separate the state concerns between managing the partition state (which partition is selected / whether the search bar is displayed, etc..

And the results displayed by the partition. I think this would reduce the complexity of this component and provide Terra UI with a nice re-usable component.

Reply:
Let me try splitting it out into a separate component - I think the terminology confuses me a little with "partition", but let me give it an initial crack and see what you think!

Reply:
Thanks! I was thinking of partition because we are viewing segments of the data. I could see a use case where we use the bar but with a different partition scheme (e.g. numbers or something like "A-D", "E-H", "I-M"). But no need to worry about other "partition" types right now. 

I don't feel strongly about the name - feel free to use something clearer. 
Other suggestions: "SegmentSelector" or "SegmentBar"